### ISSUE TITLE
5a-75B turning outputs into inputs? also 5a-75e

### ISSUE THREAD

**Issue by fanoush** [q3k/chubby75#37]
Wondered how the outputs can be turned into inputs. On the 5a-75B when changing TC74VHC245 direction pin it would probably feed 5V into FPGA? so maybe powering from 3.3V instead of  5V would solve that? Does the board work when powered from 3.3 or can the vcc to TC74VHC245 be changed to 3.3V

Thinking about interfacing it with 5V CPU bus like the C64 cartridge connector. 

BTW there is also 5a-75e https://www.aliexpress.com/item/32836510389.html having same Lattice FPGA and twice the connectors, also some listings like this one https://www.aliexpress.com/item/33010384708.html say "Wide working voltage range with DC3.3~6V." so maybe the output of that one is not 5V?

**Comment by cyber-murmel** [comment]
> A parasitic diode is formed between the bus and VCC terminals.Therefore, the VHC245 cannot be used to interface 5 V to 3 Vsystems directly.

 [Datasheet](https://www.onsemi.com/pub/Collateral/MC74VHC245-D.PDF)

Although there is a 33 Ohm resistor between the connector and the IC, I wouldn't really trust it to not put 5V on my 3.3V rail.

**Comment by cyber-murmel** [comment]
My current plan is to have some PCBs produced to replace the IC and just bridge the connection.

![image](https://user-images.githubusercontent.com/30078229/73494547-f7df4400-43b4-11ea-83d6-363963c7ed76.png)

**Comment by FFY00** [comment]
@cyber-murmel do you have the design somewhere? I was thinking of ordering some pcbs.

**Comment by cyber-murmel** [comment]
@FFY00 The problem with this particular design was that fabs generally don't accept castellated holes that small.

But I recently got around to design and install another version of this idea. This time with three plated through holes per pad to wick in / wick through the solder.
![image](https://user-images.githubusercontent.com/30078229/87224952-979ff580-c389-11ea-8005-01dbeb528a5f.png)

**Comment by FFY00** [comment]
Hum, that might be a little bit difficult to line up. Let me know when you give it a try. Right now I think I can live with hand soldered jumpers on one or two chips.

**Comment by cyber-murmel** [comment]
Most of the holes aren't clogged from the fab. I found it easy to align them by looking through the holes and having the pads of the footprint behind it create a parallax effect. Like when you line up a stencil with a PCB.

**Comment by cyber-murmel** [comment]
I also found, that a big single-flat soldering tip on powerful iron with lots of flux is easier to use, than hot air.

**Comment by tomverbeure** [comment]
@cyber-murmel Hey cyber-murmel, 
do you mind if I use that picture that you posted above in a presentation about FPGA board hacking?
Tom

**Comment by cyber-murmel** [comment]
@tomverbeure The picture above is under the [Creative Commons Attribution-NonCommercial License](https://creativecommons.org/licenses/by-nc/4.0/legalcode)
![](https://creativecommons.org/images/deed/svg/cc_white.svg) ![](https://creativecommons.org/images/deed/svg/attribution_icon_white.svg) ![](https://creativecommons.org/images/deed/svg/nc_white.svg)

**Comment by Disasm** [comment]
I came up with these flex boards from OSH Park: https://oshpark.com/shared_projects/Og8YtGBU
With minor rework they can be soldered with a regular sodering iron and some flux.

![IMG_20200830_165952](https://user-images.githubusercontent.com/1418749/91661137-bba6ca00-eae2-11ea-99b5-8d3aba21919a.jpg)
![IMG_20200825_211245](https://user-images.githubusercontent.com/1418749/91661160-e55ff100-eae2-11ea-87d1-6a43dfde9ed5.jpg)

**Comment by derekmulcahy** [comment]
Sweet, just ordered a set, thanks.

**Comment by FFY00** [comment]
Also ordered, many thanks :blush:!

**Comment by fanoush** [comment]
Thanks, I guess one just needs to not forget clicking "Flex" checkbox on ordering page (?)

**Comment by Disasm** [comment]
@fanoush yes! Non-flex boards will be useless.

**Comment by tomverbeure** [comment]
Nice! Just ordered a bunch of them as well.

**Comment by jadafi** [comment]
So just to undestand this correctly: with these bridges the inputs are directly wired to the fpga, thus they are only 3.3V tolerant?

**Comment by Disasm** [comment]
@jadafi Exactly.

**Comment by kittennbfive** [comment]
I might have found a way to get 5V tolerant inputs. I documented it on my github: https://github.com/kittennbfive/5A-75B-tools/tree/master/I_want_inputs

TL;DR: Replace (some of) the buffers with SN74LVC245APWR with some surgery to modify 2 connections on each IC.

**Comment by Disasm** [comment]
Speaking of IC replacement hacks, there is another option that involves SN74CBT3245APW bidirectional FET switches: https://twitter.com/Claude1079/status/1231194849350647808
Seems like it's also requires additional hacks, but looks relatively easy.

**Comment by eibach** [comment]
I have documented my modifications here: https://zeromips.org/posts/2022-05-29-5a-75b
Involves flex PCB as well as SN74CBT3245APW.
